File: annotation/src/main/java/im/mingxi/miko/annotation/FunctionHookEntry.java
package im.mingxi.miko.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.TYPE})
@Retention(RetentionPolicy.CLASS)
public @interface FunctionHookEntry {
    // 项目内部名称，可不填
    String itemName() default "NonName";

    // 项目类型，默认为通用项目
    int itemType() default 0;

    int COMMON_ITEM = 0;
    int WECHAT_ITEM = 1;
    int QQ_ITEM = 2;
    int TIM_ITEM = 3;
    int COMMON_QQ_TIM_ITEM = 4;
}

File: annotation/src/main/kotlin/im/mingxi/miko/annotation/FunctionHookEntry.kt
package im.mingxi.miko.annotation

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class FunctionHookEntry(
    val itemName: String = "NonName",  // 项目内部名称，可不填
    val itemType: Int = COMMON_ITEM// 项目类型，默认为通用项目
) {
    companion object {
        const val COMMON_ITEM: Int = 0
        const val WECHAT_ITEM: Int = 1
        const val QQ_ITEM: Int = 2
        const val TIM_ITEM: Int = 3
        const val COMMON_QQ_TIM_ITEM: Int = 4
    }
}

